src/Psalm/Internal/Cli/Psalm.php

```diff
@@ -421,9 +421,7 @@ private static function initShowInfo(array $options): bool
 
     private static function initIsDiff(array $options): bool
     {
-        return !isset($options['no-diff'])
-            && !isset($options['set-baseline'])
-            && !isset($options['update-baseline']);
+        return !isset($options['no-diff']);
```

Comment:
i cant really rationalize this. i feel like it hacks the caching for whatever purpose. IMHO then `--no-diff` should be passed explicitly (ref #7681)

Reply:
I fear Psalm may be unable to refresh or recreate the baseline unless it has analyzed your whole codebase. It would make sense at least. So whenever your ask it to do that, it can't just analyze files that changed or were removed and launch the whole analysis.

You may have to choose between having a squeaky clean baseline and a fast analysis

Reply:
im not sure i follow, it seems psalm is able to recreate the baseline from cache just fine :thinking: 

i can confirm after `rm psalm-baseline.xml` regenerates the same baseline file using `vendor/bin/psalm --set-baseline=psalm-baseline.xml`

Reply:
i can also confirm after `rm psalm-baseline.xml` and intentionally forcing an error (referenced an unknown class in src/ somewhere), the baseline is updated as expected

Reply:
i can also confirm after fixing that error i get `Baseline for issue "UndefinedClass" has 1 extra entry. (see https://psalm.dev/316)` AND it's removed from the baseline file

Reply:
There's two things that can reduce the time of Psalm's analysis:
- the cache (we store parsed files and also store method/class definitions)
- the diff option (Psalm will only check changed files instead of analyzing everything again)

Both can be controlled through --no-cache and --no-diff (though I think --no-cache implies --no-diff).

I think you may have misinterpreted Psalm using the --no-diff mode when you ask it to regenerate baseline vs Psalm using no cache at all.

Could you make a few tests with --no-diff, --no-cache and --set-baseline to see if it matches your expectations?